### PR TITLE
feat(graphsync): unify req & resp Pause, Unpause & Cancel by RequestID

### PR DESCRIPTION
**Pull request by rvagg**
Still working on this, it's not quite there yet, there's a problem with state management somewhere in the responsemanager that I'm having some difficulty figuring out.

### CODE REVIEW

**Comment by rvagg** [comment]
This unifies the external API into singular commands that work across requests and responses via `RequestID` and tests pass, but there's still more instances internally that couple both `peer.ID` and `RequestID` that could be dealt with so I'm going to do some minor additional cleanup to try and remove at least some of that:

```
$ git grep 'peer.ID.*RequestID\|RequestID.*peer.ID' | wc -l
33
```

**Comment by rvagg** [comment]
8c1008c removes the pair from asyncloader, _but_ that might not be appropriate, depending on how we expect users to consume the API--we seem to want to expose the peer ID to the loadattemptqueue via its `loadAttempter` callback, which we currently don't use but could theoretically be used by someone external wiring these pieces up manually. @hannahhoward is this something we want to retain? Should I back 8c1008c out of here?

**Comment by rvagg** [comment]
backed out 8c1008c because Hannah is working on a major change to the asyncloader anyway and that's going to make this change irrelevant and will make a merge more difficult

File: requestmanager/server.go [hunk 1]
```diff
@@ -233,7 +233,7 @@ func (rm *RequestManager) cancelRequest(requestID graphsync.RequestID, onTermina
 	if !ok {
 		if onTerminated != nil {
 			select {
-			case onTerminated <- graphsync.RequestNotFoundErr{}:
+			case onTerminated <- &graphsync.RequestNotFoundErr{}:
```
Comment (hannahhoward) [inline requestmanager/server.go:236]: generally I prefer not to use pointer types for errors -- they're immutable values. any reason you converted it? I see you are working with errors.Is but I don't think you should need that to make it work.

**Review by hannahhoward — COMMENTED** [review]
@rvagg this is looking really great! I'm excited to see this wasn't too hard and really excited to see responseKeys go away in the responsemanager

What isn't done? It looks pretty ok on the tests.

File: requestmanager/server.go (hunk 1 above)
Comment (rvagg) [reply requestmanager/server.go:236]: I had trouble making `errors.As` working without pointers, but Dan helped me figure that one out; so this is reverted

File: responsemanager/server.go [hunk 2]
```diff
@@ -293,8 +298,8 @@ func (rm *ResponseManager) taskDataForKey(key responseKey) queryexecutor.Respons
 }
 
 func (rm *ResponseManager) startTask(task *peertask.Task) queryexecutor.ResponseTask {
-	key := task.Topic.(responseKey)
-	taskData := rm.taskDataForKey(key)
+	key := task.Topic.(queueTopic)
+	taskData := rm.taskDataForKey(key.requestID)
 	if taskData.Empty {
 		rm.responseQueue.TaskDone(key.p, task)
```
[the remark this answers is not in the document]
Comment (rvagg) [reply responsemanager/server.go:304]: not too complicated, done, and the custom key type is **gone**

**Comment by rvagg** [comment]
good to go as far as I'm concerned

**Comment by rvagg** [comment]
codecov interestingly reporting that we apparently don't test pause or cancel from the responsemanager from the external API; I suppose this is fine, but maybe now we're unifying them that should be something to consider

**Pull request merged**